src/Core/Domain/Security/Model/SecurityPolicy.php

```diff
@@ -0,0 +1,275 @@
+<?php
+
+/*
+ * Copyright 2005 - 2021 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Domain\Security\Model;
+
+use Centreon\Domain\Common\Assertion\Assertion;
+
+class SecurityPolicy
+{
+
+    public const SPECIAL_CHARACTERS_LIST = '@$!%*?&',
+         MIN_PASSWORD_LENGTH = 8,
+         MAX_PASSWORD_LENGTH = 128,
+         MIN_ATTEMPTS = 1,
+         MAX_ATTEMPTS = 10,
+         MIN_BLOCKING_DURATION = 1,
+         MAX_BLOCKING_DURATION = 604800, // 7 days in seconds.
+         MIN_PASSWORD_EXPIRATION = 604800, // 7 days in seconds.
+         MAX_PASSWORD_EXPIRATION = 31536000, // 12 months in seconds.
+         MIN_NEW_PASSWORD_DELAY = 3600, // 1 hour in seconds.
+         MAX_NEW_PASSWORD_DELAY = 604800; // 7 days in seconds.
+
+    /**
+     * @param int $passwordMinimumLength
+     * @param bool $hasUppercase
+     * @param bool $hasLowercase
+     * @param bool $hasNumber
+     * @param bool $hasSpecialCharacter
+     * @param bool $canReusePassword
+     * @param int|null $attempts
+     * @param int|null $blockingDuration
+     * @param int|null $passwordExpiration
+     * @param int|null $delayBeforeNewPassword
+     */
+    public function __construct(
+        private int $passwordMinimumLength,
+        private bool $hasUppercase,
+        private bool $hasLowercase,
+        private bool $hasNumber,
+        private bool $hasSpecialCharacter,
+        private bool $canReusePassword,
+        private ?int $attempts = null,
+        private ?int $blockingDuration = null,
+        private ?int $passwordExpiration = null,
+        private ?int $delayBeforeNewPassword = null
```

Comment:
I guess if those can be null, it means that they are not mandatory to instanciate the class. So you can remove those from the constructor and add them as properties of your class with their getters and setters

Reply:
Agreed in many common case, but here in a business logic those properties are mandatory, even if they have a null value. this is why i add them to the constructor.